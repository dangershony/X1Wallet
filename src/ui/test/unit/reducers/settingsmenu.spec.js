import settingsReducer, { OPEN_SETTINGS, CLOSE_SETTINGS } from 'reducers/settingsmenu'

describe('reducers', () => {
  describe('settingsReducer', () => {
    it('should handle initial state', () => {
      expect(settingsReducer(undefined, {})).toMatchSnapshot()
    })

    it('should have OPEN_SETTINGS', () => {
      expect(OPEN_SETTINGS).toEqual('OPEN_SETTINGS')
    })

    it('should have CLOSE_SETTINGS', () => {
      expect(CLOSE_SETTINGS).toEqual('CLOSE_SETTINGS')
    })

    it('should correctly openSettings', () => {
      expect(
        settingsReducer(undefined, { type: OPEN_SETTINGS, settingsOpen: true })
      ).toMatchSnapshot()
    })

    it('should correctly closeSettingsMenu', () => {
      expect(
        settingsReducer(undefined, {
          type: CLOSE_SETTINGS,
          settingsOpen: false,
          activeSubMenu: null,
        })
      ).toMatchSnapshot()
    })
  })
})
