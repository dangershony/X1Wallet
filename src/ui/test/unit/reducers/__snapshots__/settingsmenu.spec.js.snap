// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers settingsReducer should correctly closeSettingsMenu 1`] = `
Object {
  "activeSubMenu": null,
  "isSettingsMenuOpen": false,
}
`;

exports[`reducers settingsReducer should correctly openSettings 1`] = `
Object {
  "activeSubMenu": null,
  "isSettingsMenuOpen": true,
}
`;

exports[`reducers settingsReducer should handle initial state 1`] = `
Object {
  "activeSubMenu": null,
  "isSettingsMenuOpen": false,
}
`;
